src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,266 @@
+    shortName = StandardArgumentDefinitions.GENOMICSDB_CALLSET_MAP_FILE_SHORT_NAME,
+    doc = "JSON file including callset to GenomicsDB row maps")
+  private String callsetMapJSONFile = "";
+
+  @Argument(fullName = StandardArgumentDefinitions.GENOMICSDB_CREATE_WORKSPACE_LONG_NAME,
+  shortName = StandardArgumentDefinitions.GENOMICSDB_CREATE_WORKSPACE_SHORT_NAME,
+  doc = "Boolean flag - true means it will create GenomicsDB workspace, otherwise not",
+  optional = true)
+  private Boolean createWorkspace = false;
+
+  @Override
+  public boolean requiresIntervals() { return true; }
+
+  @Override
+  public boolean requiresReads() { return false; }
+
+  @Override
+  public boolean requiresReference() {
+    return false;
+  }
+
+  // The GenomicsDB importer object
+  private GenomicsDBImporter importer = null;
+
+  // Intervals from command line (singleton for now)
+  private List<ChromosomeInterval> intervals;
+
+  /**
+   * From the command line validate
+   * 1. GenomicsDB workspace exists
+   * 2. GenomicsDB workspace is a valid one
+   *
+   * @return  Error messages for the exceptions
+   */
+  @Override
+  protected String[] customCommandLineValidation() {
+    String[] messages = new String[2];
+    int index = 0;
+
+    if (workspace.isEmpty()) {
+      messages[index] = " Must specify a workspace with -GW or --GenomicsDBWorkspace option";
+      return messages;
+    }
+
+    File workspaceDir = new File(workspace);
+
+    if (!createWorkspace) {
+      if (!Files.isDirectory(workspaceDir.toPath()))
+        messages[index++] = "No such directory or workspace found: " + workspace;
+
+      boolean check = new File(workspace + "/__tiledb_workspace.tdb").exists();
+
+      if (Files.isDirectory(workspaceDir.toPath()) && !check) {
+        messages[index++] = "Not a valid GenomicsDB workspace: " + workspace;
+      }
+    }
+    
+    if (sizePerColumnPartitionPerSample < 1024L) {
+      messages[index++] = "Size per column partition per sample is tool small." +
+        " Consider using larger size than 10KBytes";
+    }
+
+    return index==0 ? null : messages;
+  }
+
+  /**
+   * Before traversal, fix configuration parameters and initialize
+   * GenomicsDB. Hard-coded to handle only VCF files and headers
+   */
+  @Override
+  public void onTraversalStart() {
+
+    File workspaceDir = new File(workspace);
+
+    if (createWorkspace) {
+
+      int ret = GenomicsDBImporter.createTileDBWorkspace(workspaceDir.getAbsolutePath());
+
+      if (ret > 0) {
+        try {
+          throw new IOException("Directory " + workspaceDir + " already exists");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
+      } else if (ret < 0) {
+        try {
+          throw new IOException("Error creating GenomicsDB workspace");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
```

Comment:
Instead of throwing `IOException` and catching it immediately, throw a `UserException` instead. Make sure to pass the `IOException` to the `UserException` constructor so that it will show up in the stack trace. Use a `GATKException` for errors that are not the user's fault.